### ISSUE TITLE
Make actions available in a collection tree view

### ISSUE THREAD

**Issue by robdekort** [statamic/ideas#575]
It would be nice if actions were available in a collections tree view. Currently a user has to navigate to the list view. This makes actions not super easily discoverable for end users.

Adding them to the flyout menu - _with which you can run an action only for one entry_ - would probably be enough. This would however mean an flyout menu must be added to the root page of a (structured) collection.

**Comment by isarphilipp** [comment]
@robdekort Have you ever looked under the hood? We're having the same issue/request and I suppose it's not that complicated because all magic is already there for the list view. If we find a bit of time, we'll dig into it.

A related but separate request for us would be to have actions on the entry view.

**Comment by robdekort** [comment]
Oh I love that idea! 

I haven't looked there. It's above my pay grade I'm afraid.

**Comment by jacksleight** [comment]
Another related request would be adding folder actions to the assets grid view. At the moment you can only delete folders from the list view. Probably a bit more complicated to implement as folders open with a single click, perhaps adding something that appears on hover would work.

**Comment by mmodler** [comment]
I came to this topic via Duncans great duplicator addon. But many users only use tree view, so they never see this feature.

The PR above is unfinished, so any chance to see this added by the core team? My frontend / vue skills are too poor to contribute. Thanks!

**Comment by tobimori** [comment]
Disappointing to see this open since 2021

**Comment by jackmcdade** [comment]
> Disappointing to see this open since 2021

I wish we could tackle all the things at once, but unfortunately it's not possible. I assure you, we keep very busy working through everything we can, doing our best to keep priority and how many people are affected by the lack of a feature in mind as we go. Just because a feature request is open here doesn't mean we will or won't get to it. It's just a place to save the ideas.

**Comment by ryanmitchell** [comment]
Just for some context there are some significant differences worth noting between tree actions and listing actions which make this more complicated than it immediately appears. 

Actions in the tree dont take immediate effect, rather it requires the 'save' action.
Actions in the listing do take immediate effect.

So you either need to 'queue' actions in the tree view for when save is hit, or you need to change the tree view to take effect immediately, or you need to create a new property on action to determine if its tree only or listing only. 

All this is to say I can understand why it hasnt been done yet.

**Comment by tobimori** [comment]
I understand. I'd honestly expect them (or at least the duplicate action) to behave the same (be applied immediately).
When you "create a child entry", it is also created immediately.

**Comment by ryanmitchell** [comment]
Yep me too - but that means reconfiguring the tree view considerably as it then would need to refresh after each action.

**Comment by oktomartin** [comment]
I'd be fine with a prompt of "Pending changes, needs saving before editing underlying collection".
There probably is place for two kinds of actions, one that needs an untouched / freshly saved tree, prompting for it, with a refresh after the action, and one that operates on the current structure, e.g. sort subtree entries by column.

**Comment by Devsome** [comment]
Just noticed this on Discord, as I was looking for this in Statamic 6 and found a way to enable it without directly modifying the core or writing a vendor patcher.
